customer-terminate-service/app/src/main/java/se/inera/intyg/cts/application/service/TerminationServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+import org.springframework.data.crossstore.ChangeSetPersister.NotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+import org.springframework.web.server.ResponseStatusException;
+import se.inera.intyg.cts.application.dto.CreateTerminationDTO;
+import se.inera.intyg.cts.application.dto.TerminationDTO;
+import se.inera.intyg.cts.application.dto.TerminationDTOMapper;
+import se.inera.intyg.cts.domain.model.Termination;
+import se.inera.intyg.cts.domain.model.TerminationBuilder;
+import se.inera.intyg.cts.domain.model.TerminationId;
+import se.inera.intyg.cts.domain.model.TerminationStatus;
+import se.inera.intyg.cts.domain.repository.TerminationRepository;
+import se.inera.intyg.cts.domain.service.SendPackagePassword;
+
+@Service
+public class TerminationServiceImpl implements TerminationService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TerminationServiceImpl.class);
+
+  private final TerminationRepository terminationRepository;
+  private final SendPackagePassword sendPackagePassword;
+
+  public TerminationServiceImpl(TerminationRepository terminationRepository, SendPackagePassword sendPackagePassword) {
+    this.terminationRepository = terminationRepository;
+    this.sendPackagePassword = sendPackagePassword;
+  }
+
+  @Override
+  @Transactional
+  public TerminationDTO create(CreateTerminationDTO createTerminationDTO) {
+
+    final var termination = TerminationBuilder.getInstance()
+        .creatorHSAId(createTerminationDTO.creatorHSAId())
+        .creatorName(createTerminationDTO.creatorName())
+        .careProviderHSAId(createTerminationDTO.hsaId())
+        .careProviderOrganizationNumber(createTerminationDTO.organizationNumber())
+        .careProviderOrganizationRepresentativePersonId(createTerminationDTO.personId())
+        .careProviderOrganizationRepresentativePhoneNumber(createTerminationDTO.phoneNumber())
+        .careProviderOrganizationRepresentativeEmailAddress(createTerminationDTO.emailAddress())
+        .create();
+
+    final var createdTermination = terminationRepository.store(termination);
+    LOG.info("Created termination with id '{}' for care provider '{}'",
+        createdTermination.terminationId().id(),
+        createdTermination.careProvider().hsaId().id()
+    );
+
+    return toDTO(createdTermination);
+  }
+
+  @Override
+  @Transactional
+  public Optional<TerminationDTO> findById(UUID terminationId) {
+    final var termination = terminationRepository.findByTerminationId(
+        new TerminationId(terminationId));
+
+    return termination.map(TerminationDTOMapper::toDTO);
+  }
+
+  @Override
+  @Transactional
+  public List<TerminationDTO> findAll() {
+    return terminationRepository.findAll().stream()
+        .map(TerminationDTOMapper::toDTO)
+        .collect(Collectors.toList());
+  }
+
+  /**
```

Comment:
Javadoc that doesn't add any additional value/description than the method name itself is redundant and create noise. I think your code is self-documenting.

Reply:
I prefere javadoc as best practise and have never felt that it create noise. All external libraries does contain it and i find it helpfull that you can get a small description just by howering over the classname or method. By experience self-documented methods tends to not be self-documented in the long run and/or be self documented to the one who writes them. :)